Executables/Examples/example_cli.cpp

```diff
@@ -0,0 +1,126 @@
+#include "NSL.hpp"
+
+/*
+ * This example shows a default use case of the cli following the used library:
+ * https://github.com/CLIUtils/CLI11
+ *
+ * Or directly look for a simple example:
+ * https://github.com/CLIUtils/CLI11/blob/main/examples/simple.cpp
+ * */
+
+void printMyParams(std::string file, int runParamA, double runParamB, NSL::complex<double> runParamC){
+    NSL::Logger::info("printMyParams: working with parameter file: {}", file);
+    NSL::Logger::info("printMyParams: using run parameter A: {}", runParamA);
+    NSL::Logger::info("printMyParams: using run parameter B: {}", runParamB);
+    NSL::Logger::info("printMyParams: using run parameter C: {}", NSL::to_string(runParamC));
+}
+
+int main(int argc, char ** argv){
+    // This is a very compact form to initialize the NSL::Logger as well as
+    // a parameter class from the command line.
+    // Basically, the command line is parsed for the parameters in the first
+    // initializer list, i.e. it is looking for 
+    // --file STR
+    // --runParamA INT
+    // --runParamB DOUBLE
+    // --runParamC COMPLEX
+    // the data types are provided with the template parameters matching 
+    // the order of the parameter strings. 
+    // The second initializer list provides a set of help messages that 
+    // are displayed using --help or -h. If omitted messages will be empty.
+    // The last argument (optional) provides a name to the executable which
+    // is printed for the help message
+    NSL::Parameter params = NSL::init<std::string, int, double, NSL::complex<double>>(
```

Comment:
This makes me think: must the person calling the CLI specify every single option?  Or can the programmer set a default (µ=0, for example, being a natural one).

Reply:
the types and strings need to be there in order to define the parameter class and options. I would have to try, but I think I set the cli to use optional arguments only, thus if µ is not specified when executing the code it will be set to 0 be default.

Reply:
With the physics being in the input file and the runtime parameters being flags I think my question is moot anyway.